src/gui/attributetable/qgsattributetabledelegate.cpp

```diff
@@ -93,6 +93,13 @@ void QgsAttributeTableDelegate::setModelData( QWidget *editor, QAbstractItemMode
   if ( !eww )
     return;
 
+  // This fixes https://issues.qgis.org/issues/16492
+  if ( ! vl->getFeature( fid ).isValid( ) )
```

Comment:
For speed you could tighten up this request a lot - fetch no geometry or attributes.